components/DragAndDropInput.vue

```diff
@@ -0,0 +1,267 @@
+<script setup lang="ts">
+import { computed } from "vue";
+
+const props = defineProps<{
+  maxFilesNumber: number;
+  maxFileSizeMB: number;
+}>();
+
+const isDragActive = ref(false);
+const message = ref("");
+const uploadedFiles = ref<string[]>([]);
+const notUploadedFiles = ref<string[]>([]);
+const root = ref<HTMLElement | null>(null);
+
+const maxFileSizeBytes = computed(() => {
+  return props.maxFileSizeMB * 1000000;
+});
+
+const resetState = () => {
+  message.value = "";
+  uploadedFiles.value = [];
+  notUploadedFiles.value = [];
+};
+
+const handleDrag = (e: DragEvent) => {
+  if (e.type === "dragenter" || e.type === "dragover") {
+    isDragActive.value = true;
+    resetState();
+  }
+  if (e.type === "dragleave") {
+    isDragActive.value = false;
+  }
+};
+
+const numberOfFilesExceeded = (
+  listOfFiles: FileList | DataTransferItemList
+) => {
+  if (listOfFiles.length > props.maxFilesNumber) {
+    message.value = `You can upload up to ${props.maxFilesNumber} files at a time!`;
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const handleDrop = (e: DragEvent) => {
+  e.stopPropagation();
+  isDragActive.value = false;
+
+  if (e.dataTransfer?.items) {
+    if (numberOfFilesExceeded(e.dataTransfer.items)) return;
+    [...e.dataTransfer.items].forEach((item) => {
+      if (item.kind === "file") {
+        const file = item.getAsFile();
+        if (file) {
+          if (file.size < maxFileSizeBytes.value) {
+            uploadedFiles.value.push(file.name);
+            // send to backend
+          } else {
+            notUploadedFiles.value.push(file.name);
+          }
+        }
+      }
+    });
+  } else if (e.dataTransfer?.files) {
+    if (numberOfFilesExceeded(e.dataTransfer.files)) return;
+    [...e.dataTransfer.files].forEach((file) => {
+      if (file.size < maxFileSizeBytes.value) {
+        uploadedFiles.value.push(file.name);
+        // send to backend
+      } else {
+        notUploadedFiles.value.push(file.name);
+      }
+    });
+  }
+};
+
+const handleUpload = (e: Event) => {
+  const target = e.target as HTMLInputElement;
+  if (!target.files || numberOfFilesExceeded(target.files)) return;
+  [...target.files].forEach((file) => {
+    if (file.size < maxFileSizeBytes.value) {
+      uploadedFiles.value.push(file.name);
+      // send to backend
+    } else {
+      notUploadedFiles.value.push(file.name);
+    }
+  });
+};
+
+const handleKeydown = (e: KeyboardEvent) => {
+  const input = root.value?.querySelector("#file") as HTMLInputElement | null;
+  if (e.code === "Enter") {
+    resetState();
+    input?.click();
+  }
+};
+</script>
+
+<template>
+  <div ref="root" class="wrapper">
```

Comment:
let's try to use BEM for classes in the component 😄 you can assume that the block (B in BEM) is the root element in the component - in this case first `div` and rest elements will be elements (😅 ) of this block with modifiers if needed (I saw there is at least one modifier in the template for `isDragActive` state) 👌